pkg/controller/schedule_controller.go

```diff
@@ -197,22 +213,24 @@ func (controller *scheduleController) processNextWorkItem() bool {
 }
 
 func (controller *scheduleController) processSchedule(key string) error {
-	glog.V(4).Infof("processSchedule for key %q", key)
+	logContext := controller.logger.WithField("key", key)
+
+	logContext.Debug("Running processSchedule")
 	ns, name, err := cache.SplitMetaNamespaceKey(key)
 	if err != nil {
-		glog.V(4).Infof("error splitting key %q: %v", key, err)
+		logContext.WithField("error", err).Info("Error splitting queue key")
 		return err
 	}
 
-	glog.V(4).Infof("Getting schedule %s", key)
+	logContext.Debug("Getting Schedule")
 	schedule, err := controller.schedulesLister.Schedules(ns).Get(name)
 	if err != nil {
 		// schedule no longer exists
 		if apierrors.IsNotFound(err) {
-			glog.V(4).Infof("schedule %s not found: %v", key, err)
+			logContext.WithField("error", err).Info("Schedule not found")
 			return nil
 		}
-		glog.V(4).Infof("error getting schedule %s: %v", key, err)
+		logContext.WithField("error", err).Info("Error getting schedule")
```

Comment:
Please don't log an error at Info level, please don't log and return an error, just return it.